File: arrays/arrays.go
package main

import (
	"fmt"
	"time"
)

func main() {
	sum := 0
	start := time.Now()

	for e := 0; e < 30; e++ {
		x := make([]int, 0, 1000000)
		for i := 0; i < 1000000; i++ {
			x = append(x, i)
		}

		y := make([]int, 0, 1000000-1)
		for i := 0; i < 1000000-1; i++ {
			y = append(y, x[i]+x[i+1])
		}

		sum = 0
		for i := 0; i < 1000000; i += 100 {
			sum += y[i]
		}
	}

	elapsed := time.Since(start)

	fmt.Printf("Elapsed (sec) %s\n", elapsed)
	fmt.Println(sum)
}

File: arrays/arrays.py
import time

sum = 0
start = time.time()
for e in xrange(30):
    sum = 0

    x = []
    for i in xrange(1000000):
        x.append(i)

    y = []
    for i in xrange(1000000 - 1):
        y.append(x[i] + x[i+1])

    for i in xrange(0, 1000000,  100):
        sum +=  y[i]

end = time.time()
print('Elapsed ' + str(end-start))
print sum
